azkaban-common/src/main/java/azkaban/jobcallback/JobCallbackValidator.java

```diff
@@ -0,0 +1,121 @@
+package azkaban.jobcallback;
+
+import static azkaban.jobcallback.JobCallbackConstants.DEFAULT_POST_BODY_LENGTH;
+import static azkaban.jobcallback.JobCallbackConstants.HTTP_GET;
+import static azkaban.jobcallback.JobCallbackConstants.HTTP_POST;
+import static azkaban.jobcallback.JobCallbackConstants.JOB_CALLBACK_BODY_TEMPLATE;
+import static azkaban.jobcallback.JobCallbackConstants.JOB_CALLBACK_REQUEST_METHOD_TEMPLATE;
+import static azkaban.jobcallback.JobCallbackConstants.JOB_CALLBACK_URL_TEMPLATE;
+import static azkaban.jobcallback.JobCallbackConstants.MAX_POST_BODY_LENGTH_PROPERTY_KEY;
+import static azkaban.jobcallback.JobCallbackConstants.SEQUENCE_TOKEN;
+import static azkaban.jobcallback.JobCallbackConstants.STATUS_TOKEN;
+
+import java.util.Collection;
+
+import org.apache.log4j.Logger;
+
+import azkaban.utils.Props;
+
+/**
+ * Responsible for validating the job callback related properties at project
+ * upload time
+ * 
+ * @author hluu
+ *
+ */
+public class JobCallbackValidator {
+
+  private static final Logger logger = Logger
+      .getLogger(JobCallbackValidator.class);
+
+  /**
+   * Make sure all the job callback related properties are valid
+   * 
+   * @param jobProps
+   * @param error
+   * @return number of valid job callback properties. Mainly for testing
+   *         purpose.
+   */
+  public static int validate(String jobName, Props serverProps, Props jobProps,
+      Collection<String> errors) {
+    int maxNumCallback =
+        serverProps.getInt(
+            JobCallbackConstants.MAX_CALLBACK_COUNT_PROPERTY_KEY,
+            JobCallbackConstants.DEFAULT_MAX_CALLBACK_COUNT);
+
+    int maxPostBodyLength =
+        serverProps.getInt(MAX_POST_BODY_LENGTH_PROPERTY_KEY,
+            DEFAULT_POST_BODY_LENGTH);
+
+    int totalCallbackCount = 0;
+    for (JobCallbackStatusEnum jobStatus : JobCallbackStatusEnum.values()) {
+      totalCallbackCount +=
+          validateBasedOnStatus(jobProps, errors, jobStatus, maxNumCallback,
+              maxPostBodyLength);
+    }
+
+    logger.info("Found " + totalCallbackCount + " job callbacks for job "
+        + jobName);
+    return totalCallbackCount;
+  }
+
+  private static int validateBasedOnStatus(Props jobProps,
+      Collection<String> errors, JobCallbackStatusEnum jobStatus,
+      int maxNumCallback, int maxPostBodyLength) {
+
+    int callbackCount = 0;
+    // replace property templates with status
+    String jobCallBackUrl =
+        JOB_CALLBACK_URL_TEMPLATE.replaceFirst(STATUS_TOKEN, jobStatus.name()
+            .toLowerCase());
+
+    String requestMethod =
+        JOB_CALLBACK_REQUEST_METHOD_TEMPLATE.replaceFirst(STATUS_TOKEN,
+            jobStatus.name().toLowerCase());
+
+    String httpBody =
+        JOB_CALLBACK_BODY_TEMPLATE.replaceFirst(STATUS_TOKEN, jobStatus.name()
+            .toLowerCase());
+
+    for (int i = 1; i <= maxNumCallback; i++) {
```

Comment:
I can imagine some user starts the sequence from 0. Maybe some error checking for it?